app/models/miq_worker/runner.rb

```diff
@@ -2,6 +2,9 @@
 require 'thread'
 
 class MiqWorker::Runner
+  class ProviderUnreachable < StandardError
```

Comment:
I think I'd prefer something more generic-sounding here... `TemporaryFailure`? If it's beneficial, `ProviderUnreachable` might be a subclass of that, defined closer to where we raise it... but the general re-raise / rescue seems uninterested in such detail.